### PR TITLE
210 NPC World Spawn Dumb via region

### PR DESCRIPTION
**Pull request by DrHenny**
I made a Runelite plugin to dump NPC spawns based on region ID, so far I managed to dump these :)))

- New classes to support loading per region with our format

### CODE REVIEW

File: ElvargServer/src/main/java/com/elvarg/game/definition/loader/impl/JsonNPC.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package com.elvarg.game.definition.loader.impl;
```
Comment (tobywisener) [inline ElvargServer/src/main/java/com/elvarg/game/definition/loader/impl/JsonNPC.java:1]: I think we should replace NpcSpawnDefinition.java with this class. Also it belongs in the definition package not definition.loader.impl.

Lets be confident in our work and not have two Npc spawn loading systems

File: ElvargServer/data/saves/spawns/10028.json
Comment (tobywisener) [inline ElvargServer/data/saves/spawns/10028.json:1]: Lets move ```data/saves/spawns/``` to ```data/definitions/npc_spawns/```

File: ElvargServer/src/main/java/com/elvarg/game/definition/loader/impl/NPCSpawnLoader.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package com.elvarg.game.definition.loader.impl;
```
Comment (tobywisener) [inline ElvargServer/src/main/java/com/elvarg/game/definition/loader/impl/NPCSpawnLoader.java:1]: Replace NpcSpawnDefintionLoader.java with this class please. As per last comment we shouldn't have two separate spawn loading systems in the server at once. Delete old npc_spawns.json too

**Review by tobywisener — CHANGES_REQUESTED** [review]
Good work man, just a few logical changes then its ready for merge imho

File: ElvargServer/src/main/java/com/elvarg/game/entity/impl/npc/NPC.java [hunk 3]
```diff
@@ -475,4 +475,9 @@ public static void initImplementations(List<? extends Class<?>> implementationCl
 
 				}).collect(Collectors.toMap(ImmutablePair::getLeft, ImmutablePair::getRight));
 	}
+
+	public String description;
```
[the remark this answers is not in the document]
Comment (DrHenny) [reply ElvargServer/src/main/java/com/elvarg/game/entity/impl/npc/NPC.java:479]: It's not used currently, usually I would add the author to the description but you made me change it.

File: ElvargServer/src/main/java/com/elvarg/game/definition/loader/impl/NPCSpawnLoader.java (hunk 2 above)
Comment (DrHenny) [reply ElvargServer/src/main/java/com/elvarg/game/definition/loader/impl/NPCSpawnLoader.java:1]: Yehh, I could. I didn't just for the testing purposes.

File: ElvargServer/src/main/java/com/elvarg/game/definition/loader/impl/JsonNPC.java (hunk 1 above)
Comment (DrHenny) [reply ElvargServer/src/main/java/com/elvarg/game/definition/loader/impl/JsonNPC.java:1]: yes

**Pull request merged**